### PR TITLE
Feature/s3 upgrade

### PR DESCRIPTION
**Pull request by jbrown-xentity**
s3 filestore local integration with s3 testing system ([localstack](https://helibertoarias.com/blog/aws/localstack-locally-test-aws-docker/)).
System works in general, and integration tests are added. Unfortunately we cannot test datapusher in conjunction with s3, because of the way datasets are saved in the ckan system and how they are passed to datapusher (it is expected that the `ckan.site_url` setting in production.ini is a public URL, and when it is a local URL [localhost for dev development] then datapusher fails to download the data).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -43,5 +43,31 @@ services:
   redis:
     build: ./redis
 
+  localstack-container:
+    container_name: "localstack-container"
+    privileged: true
+    image: localstack/localstack:0.11.0
+    ports:
+      - "4566-4599:4566-4599"
+      - "8081:8081"
```
Comment (adborden) [inline docker-compose.yml:52]: Do we need all these ports published to the host? Based on the CKAN config it looks like 4572 is the only one that needs to be accessible by the host.

File: requirements.txt [hunk 2]
```diff
@@ -4,6 +4,7 @@
 -e git+https://github.com/GSA/ckanext-datajson.git@master#egg=ckanext-datajson
 -e git+https://github.com/GSA/ckanext-saml2.git@max#egg=ckanext-saml2
 -e git+https://github.com/GSA/webob.git@ckan-patch#egg=webob
+-e git+https://github.com/GSA/ckanext-s3filestore@okfn-master#egg=ckanext-s3filestore
```
Comment (adborden) [inline requirements.txt:7]: Can we name the branch off of the version? e.g. datagov-v1.0.0

**Review by adborden — COMMENTED** [review]
:+1: I think this is on the right track. BATS also has [setup/teardown methods](https://github.com/bats-core/bats-core#setup-and-teardown-pre--and-post-test-hooks) where we can put test setup and cleanup and have better isolation between tests.

**Comment by jbrown-xentity** [comment]
> 👍 I think this is on the right track. BATS also has [setup/teardown methods](https://github.com/bats-core/bats-core#setup-and-teardown-pre--and-post-test-hooks) where we can put test setup and cleanup and have better isolation between tests.

I wanted to use the [setup_file system](https://github.com/bats-core/bats-core#setup-and-teardown-pre--and-post-test-hooks), but apparently that is a newly added feature that is not released on docker. After evaluating options, it seems like each test is actually "testing" various components, even though it is adding and removing content, so I'm going to leave mostly as is for now (I cleaned up a few things to utilize common functions more).

File: docker-compose.yml (hunk 1 above)
Comment (adborden) [reply docker-compose.yml:52]: Did you want to add a code comment about this?

**Review by adborden — APPROVED** [review]

**Pull request merged**